clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1797,7 +1842,22 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     Expr *ILE = new (SemaRef.getASTContext())
         InitListExpr(SemaRef.getASTContext(), SourceLocation(), ArrayInitExprs,
                      SourceLocation());
-    ILE->setType(FD->getType());
+
+    // We need to find the type of the element for which we are generating the
+    // InitListExpr. For example, for a multi-dimensional array say a[2][3][2],
+    // the types for InitListExpr of the array and its 'sub-arrays' are -
+    // int [2][3][2], int [3][2] and int [2]. This loop is used to obtain this
```

Comment:
Do not use same numbers, it is too confusing.
I found using prime number is even better in the case of concrete computations.
For example here `int [2]` is not instantaneously obvious which `[2]` we are talking about.
Use for example `int [5][3][2]`